src/python/foglamp/device/ingest.py

```diff
@@ -53,62 +55,185 @@ class Ingest(object):
     _discarded_readings = 0  # type: int
     """Number of readings rejected before statistics were flushed to storage"""
 
-    _write_statistics_loop_task = None  # type: asyncio.Task
-    """asyncio task for :meth:`_write_statistics_loop`"""
+    _write_statistics_task = None  # type: asyncio.Task
+    """asyncio task for :meth:`_write_statistics`"""
 
-    _sleep_task = None  # type: asyncio.Task
+    _write_statistics_sleep_task = None  # type: asyncio.Task
     """asyncio task for asyncio.sleep"""
 
-    _stop = False  # type: bool
+    _stop = False
     """Set to true when the server needs to stop"""
 
+    _started = False
+    """True when the server has been started"""
+
+    _engine = None  # type: aiopg.sa.Engine
+    """Database connection pool"""
+
+    _insert_queue = None  # type: asyncio.Queue
```

Comment:
Protection isn't needed because this is a single-threaded app. That doesn't mean there can't be gnarly reentrancy though, but reentrancy is controllable since it can only happen if "await" is invoked on something that actually waits. https://stackoverflow.com/questions/45427152/does-await-guarantee-execution-order